Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/Events.proto

```diff
@@ -22,31 +34,845 @@ message EventTypeQuery{
+
+    bool MaskUntrackedAccesses = 18;
+
+    bool NormalizeReadTimestamps = 19;
+
+    bool PipWarningsPromotedToErrors = 20;
+
+    bool ValidateDistribution = 21;
+
+    string RequiredKextVersionNumber = 22;
+}
+
+// The DependencyViolationReported Event message
+message DependencyViolationReportedEvent_XLGpp{
+    string UUID = 99;
+
+    uint32 WorkerID = 1;
+
+    uint32 ViolatorPipID = 2;
+
+    uint32 RelatedPipID = 3;
+
+    FileMonitoringViolationAnalyzer_DependencyViolationType_XLGpp ViolationType = 4;
+
+    FileMonitoringViolationAnalyzer_AccessLevel_XLGpp AccessLevel = 5;
+
+    AbsolutePath_XLGpp Path = 6;
 }
+
+// The PipExecutionStepPerformanceReported Event message
+message PipExecutionStepPerformanceReportedEvent_XLGpp{
+    string UUID = 99;
+
+    uint32 WorkerID = 1;
+
+    uint32 PipID = 2;
+
+    google.protobuf.Timestamp StartTime = 3;
+
+    google.protobuf.Duration Duration = 4;
+
+    PipExecutionStep_XLGpp Step = 5;
+
+    WorkDispatcher_DispatcherKind_XLGpp Dispatcher = 6;
+}
+
+// The StatusReported Event message
+message StatusReportedEvent_XLGpp{
+    string UUID = 99;
+
+    uint32 WorkerID = 1;
+
+    google.protobuf.Timestamp Time = 2;
+
+    int32 CpuPercent = 3;
+
+    repeated int32 DiskPercents = 4;
+    
+    repeated int32 DiskQueueDepths = 5;
+
+    int32 RamPercent = 6;
+
+    int32 MachineRamUtilizationMB = 7;
+
+    int32 CommitPercent = 8;
+
+    int32 CommitTotalMB = 9;
+
+    int32 ProcessCpuPercent = 10;
+
+    int32 ProcessWorkingSetMB = 11;
+
+    int32 CpuWaiting = 12;
+
+    
+    int32 CpuRunning = 13;
+
+    int32 IoCurrentMax = 14;
+
+    int32 IoWaiting = 15;
+
+    int32 IoRunning = 16;
+
+    int32 LookupWaiting = 17;
+
+    int32 LookupRunning = 18;
+
+    int32 ExternalProcesses = 19;
+
+    repeated int64 PipsSucceededAllTypes = 20;
+
+    ExecutionSampler_LimitingResource_XLGpp LimitingResource = 21;
+
+    int32 UnresponsivenessFactor = 22;
+
+    int64 ProcessPipsPending = 23;
+
+    int64 ProcessPipsAllocatedSlots = 24;
+}
+
+// The ProcessFingerprintComputation Event message
+message ProcessFingerprintComputationEvent_XLGpp{
+    string UUID = 99;
+
+    uint32 WorkerID = 1;
+    
+    FingerprintComputationKind_XLGpp Kind = 2;
+
+    uint32 PipID = 3;
+    
+    WeakContentFingerPrint_XLGpp WeakFingerprint = 4;
+
+    repeated ProcessStrongFingerprintComputationData_XLGpp StrongFingerprintComputations = 5;
+}
+
+// The PipCacheMiss Event message
+message PipCacheMissEvent_XLGpp{
+    string UUID = 99;
+
+    uint32 WorkerID = 1;
+
+    uint32 PipID = 2;
+
+    PipCacheMissType_XLGpp CacheMissType = 3;
+}
+
+// The PipExecutionDirectoryOutputs Event message
+message PipExecutionDirectoryOutputsEvent_XLGpp{
+    string UUID = 99;
+
+    uint32 WorkerID = 1;
+
+    repeated DirectoryOutput_XLGpp DirectoryOutput = 2;
+}
+
+// The BXLInvocation Event message
+message BXLInvocationEvent_XLGpp{
+    // Unique identifier for the event - required, TODO: Remove as it may not be needed.

```

Comment:
>// Unique identifier for the event - required, TODO: Remove as it may not be needed. [](start = 4, length = 84)

you have this comment here, but not in other places. I think it should be either place next to each UUID field, or nuked from this place.